dotCMS/src/main/java/com/dotmarketing/business/PermissionBitFactoryImpl.java

```diff
@@ -2310,6 +2318,58 @@ private void deleteInsertPermission(Permissionable permissionable, String type,
         }
     }
 
+    // Method to Insert a new Permission Reference, or if it exists, it will udpate it
+    private void upsertPermission(DotConnect dc1, String permissionId, Permissionable newReference, String type) throws DotDataException {
+		if (DbConnectionFactory.isH2()) {
```

Comment:
Same principle of a command to generate the sql based on the current db type, see below comment